contracts/btc-staking/src/validation/full.rs

```diff
@@ -37,49 +29,8 @@ pub enum FullValidationError {
 }
 
 /// Verifies the new finality provider data.
-pub fn verify_new_fp(new_fp: &NewFinalityProvider) -> Result<(), ContractError> {
-    let fp_pk = verifying_key_from_hex(&new_fp.btc_pk_hex)?;
-
-    // FIXME: parameterise `bbn` prefix
-    let fp_address = to_canonical_addr(&new_fp.addr, "bbn")?;
-
-    let fp_pop = new_fp
-        .pop
-        .as_ref()
-        .ok_or(FullValidationError::MissingProofOfPossession)?;
-
-    verify_pop(&fp_pk, fp_address, fp_pop)?;
-
-    Ok(())
-}
-
-/// Verifies the proof of possession of the given address.
-fn verify_pop(
-    btc_pk: &VerifyingKey,
-    address: CanonicalAddr,
-    pop: &ProofOfPossessionBtc,
-) -> Result<(), ContractError> {
-    // get signed msg, i.e., the hash of the canonicalised address
-    let msg_hash: [u8; 32] = Sha256::digest(address.as_slice()).into();
-
-    let btc_sig_type =
-        BTCSigType::try_from(pop.btc_sig_type).map_err(FullValidationError::InvalidBtcSigType)?;
-
-    match btc_sig_type {
-        BTCSigType::BIP340 => {
-            let pop_sig = Signature::try_from(pop.btc_sig.as_slice())?;
-            verify_digest(btc_pk, &msg_hash, &pop_sig)?;
-        }
-        BTCSigType::BIP322 => {
-            // TODO?: implement BIP322 verification (#7.0)
-            return Ok(());
-        }
-        BTCSigType::ECDSA => {
-            // TODO?: implement ECDSA verification (#7.0)
-            return Ok(());
-        }
-    }
-
+/// TODO: Implement validation logic for the new finality provider.
+pub fn verify_new_fp(_new_fp: &NewFinalityProvider) -> Result<(), ContractError> {
```

Comment:
Why not leave the FP key and staker address validation?

What are the requirements for validating new FPs on the BSN side?